GPTutils.py

```diff
@@ -0,0 +1,262 @@
+    "py": "python",
+    "js": "js",
+    "jsx": "js",
+    "ts": "ts",
+    "java": "java",
+    "html": "html",
+    "md": "markdown",
+}
+
+
+class GPTWrapper:
+    def __init__(self, model: str = GPTsettings.MODEL) -> None:
+        """
+        Initializes a new instance of the GPTWrapper class.
+
+        Args:
+            model (str, optional): The name of the GPT model to use. Defaults to "text-davinci-002".
+        """
+        self.client = OpenAI()
+        self.model: str = model
+        self.conversation: list[
+            dict[str, str]
+        ] = []  # This will hold our conversation messages
+
+    def add_message(self, role: str, content: str) -> None:
+        """
+        Adds a message to the conversation history.
+
+        Args:
+            role (str): Indicates who is acting on the conversation [system, user, assistant].
+            content (str): The actual message.
+        """
+        self.conversation.append({"role": role, "content": content})
+
+    def edit_message(self, index: int, role: str, content: str) -> None:
+        """
+        Edits a message in the conversation by index.
+
+        Args:
+            index (int): The index of the message to edit.
+            role (str): Indicates who is acting on the conversation [system, user, assistant].
+            content (str): The new content of the message.
+
+        Raises:
+            IndexError: If the index is out of range.
+        """
+        if index < len(self.conversation):
+            self.conversation[index] = {"role": role, "content": content}
+        else:
+            raise IndexError("Message index out of range")
+
+    def get_response(self, max_tokens: int = 1000) -> str:
+        """
+        Gets a response from the API based on the current conversation.
+
+        Args:
+            max_tokens (int, optional): The maximum number of tokens to generate in the response. Defaults to 1000.
+
+        Returns:
+            str: The generated response.
+        """
+        response = self.client.chat.completions.create(
+            model=self.model, messages=self.conversation, max_tokens=max_tokens
+        )
+
+        return response.choices[0].message.content
+
+    def __getitem__(self, index: int) -> dict[str, str]:
+        """
+        Gets a message from the conversation by index.
+
+        Args:
+            index (int): The index of the message to get.
+
+        Returns:
+            dict[str, str]: The message at the specified index.
+        """
+        return self.conversation[index]
+
+    def __setitem__(self, index: int, value: tuple[str, str]) -> None:
+        """
+        Sets a message in the conversation by index.
+
+        Args:
+            index (int): The index of the message to set.
+            value (tuple[str, str]): A tuple containing the role and content of the new message.
+        """
+        role, content = value
+        self.edit_message(index, role, content)
+
+    def __str__(self) -> str:
```

Comment:
Please add docstrings for all new functions and classes, similar to the provided example, to maintain consistency and thorough documentation throughout the project.
```suggestion
        def __str__(self) -> str:
            """
            Returns a readable string representation of the conversation.

            Returns:
                str: A string representation of the conversation.
            """
```
        Please apply this suggestion across all new methods and classes introduced in this PR.